### PR TITLE
Fix viewer.center_on for dithered images

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to fix https://github.com/spacetelescope/jdaviz/issues/1927 .

I also have to allow centering on off-limit pixel locations to accommodate dithered images because the underlying calls all assume reference image coordinates frame.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by kecnry — COMMENTED** [review]
I agree that this is probably more intuitive (and consistent with the mouseover behavior) than assuming the reference layer.

When using monochrome layering though, it's still not incredibly obvious and an extension to pass the `data_label` (still defaulting to top-layer) could be useful - but would also add need for more test cases, so might not be worth the effort?

Either way - I think being very explicitly in the API docs for `center_on` would be worth the extra text.  Would it be worth creating a method on the viewer to access the _data label_ of the top-image, just so we have somewhere to point the user besides the mouseover display to confirm which layer will be used by default?

**Comment by pllim** [comment]
> an extension to pass the data_label (still defaulting to top-layer) could be useful 

This would violate the portability standard set by Astrowidgets upstream. I cannot change the established public API.

>  API docs

You mean like changing "Centers the view on a particular point." to "Centers the view on a particular point on the top visible layer." ❓ 

> Would it be worth creating a method on the viewer to access the data label of the top-image

Maaaybe... but it is out of scope here?

**Comment by kecnry** [comment]
> This would violate the portability standard set by Astrowidgets upstream. I cannot change the established public API.

Even if it's an optional kwarg?  No comment.

> You mean like changing "Centers the view on a particular point." to "Centers the view on a particular point on the top visible layer." 

Yes something like that.

>> Would it be worth creating a method on the viewer to access the data label of the top-image

> Maaaybe... but it is out of scope here?

Sure, we can create a follow-up PR, I was just thinking it would be something to point to in the API docstring above (instead of just saying "top visible" there would actually be somewhere to point to for the user to be able to determine _which_ layer that is.

**Comment by pllim** [comment]
> Even if it's an optional kwarg? No comment.

Yes, because if someone passes that keyword into Jdaviz, it works. But the same call would crash for other backends.

> something to point to in the API docstring above

Hmm, ok. Lemme see if I can whip up something quick.

**Review by javerbukh — APPROVED** [review]
Works as expected, thanks!

File: jdaviz/configs/imviz/plugins/viewers.py [hunk 1]
```diff
@@ -222,6 +222,17 @@ def on_limits_change(self, *args):
             return
         self.set_compass(self.state.layers[i].layer)
 
+    @property
+    def top_visible_data_label(self):
```
Comment (pllim) [inline jdaviz/configs/imviz/plugins/viewers.py:226]: @kecnry like this? I also put this in the `center_on` docstring below.

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/viewers.py:226]: Perfect, thanks!

**Review by kecnry — APPROVED** [review]

File: jdaviz/configs/imviz/plugins/viewers.py [hunk 2]
```diff
@@ -222,6 +222,17 @@ def on_limits_change(self, *args):
...
+        """Data label of the top visible layer in the viewer."""
+        try:
+            i = get_top_layer_index(self)
+        except IndexError:
+            data_label = ''
```
Comment (kecnry) [inline jdaviz/configs/imviz/plugins/viewers.py:231]: probably doesn't need coverage?

Comment (pllim) [reply jdaviz/configs/imviz/plugins/viewers.py:231]: Doesn't hurt. I added a commit to cover this line, hopefully. Thanks for catching it!

**Comment by pllim** [comment]
Thanks for the reviews!

**Comment by pllim** [comment]
This is no longer backported. I moved the change log in `main` in 088363263391280f0c2a063e35e942dfcb797995 .

**Pull request merged**